### ISSUE TITLE
kendo-menu and NavigationService: cannot read property toggleActive of undefined

### ISSUE THREAD

**Issue by fthuin** [telerik/kendo-angular#2859]
**Describe the bug**

When changing the items of a kendo-menu (for example because you have a search to filter the items), there is an error in the console. It doesn't break the component but it is an unexpected behavior.

**To Reproduce**

- Change the list of items for the `kendo-menu` after the first initialization and remove some of the items.
- For example, in this stackblitz [https://stackblitz.com/edit/angular-pefbqg](https://stackblitz.com/edit/angular-pefbqg), type `2` in the input then delete to get back to blank.

```
ERROR TypeError: Cannot read property 'toggleActive' of undefined
    at e.updateActive (main.js:5)
    at t.ngAfterViewChecked (main.js:5)
    at callProviderLifecycles (provider.ts:572)
    at callElementProvidersLifecycles (provider.ts:541)
    at callLifecycleHooksChildrenFirst (provider.ts:529)
    at checkAndUpdateView (view.ts:375)
    at callViewAction (view.ts:615)
    at execComponentViewsAction (view.ts:559)
    at checkAndUpdateView (view.ts:370)
    at callWithDebugContext (services.ts:630)
```

Technical details

```typescript
    updateActive() {
        if (!this.activeItem && this.items.hasItems) {
            const firstItem = this.items.get(DEFAULT_ACTIVE);
            firstItem.toggleActive(true);
            this.ngZone.runOutsideAngular(() => {
                resolvedPromise.then(() => {
                    this.activeIndex = DEFAULT_ACTIVE;
                });
            });
        }
    }
```

The `DEFAULT_ACTIVE` value should be `0` but it's in fact `'0'`, so accessing the array with `.get()` index `'0'` returns undefined; whereas `get(0)` would work.

**Expected behavior**

- No error should be shown and DEFAULT_ACTIVE should be `0`.

**Browser**
 - OS: Linux Mint 19
 - Browser: Chrome
 - Version 80

**Comment by dtopalov** [comment]
Fix is available in the latest [dev build](https://www.telerik.com/kendo-angular-ui/components/installation/development-builds/).

**Comment by tsvetomir** [comment]
Published in v2.0.4
